torch/distributed/nn/functional.py

```diff
@@ -219,18 +258,26 @@ def forward(ctx, group, tensor):
 
     @staticmethod
     def backward(ctx, *grad_outputs):
-        gxs = _AlltoAll.apply(ctx.group, *grad_outputs)
+        gxs = _AlltoAll.apply(ctx.group, None, *grad_outputs)
         gx = torch.sum(torch.stack(gxs), dim=0)
         return (None, gx)
 
 
 class _AlltoAll(Function):
     @staticmethod
-    def forward(ctx, group, *tensors):
+    def forward(ctx, group, out_tensor_list, *tensors):
         ctx.group = group
-        out_tensor_list = [
-            torch.empty_like(tensors[i]) for i in range(dist.get_world_size(group=group))
+        ctx.input_tensor_list = [
+            torch.empty_like(tensors[i])
+            for i in range(dist.get_world_size(group=group))
         ]
+        if out_tensor_list is None:
```

Comment:
I'm assuming we do this for BC reasons since the distributed_c10d.py all_to_all doesn't support this as being `None`?

Reply:
Yes, maybe we should make this consistent with `distributed_c10d.py`?

Reply:
Based on @mrshenli's comment I think it is fine to make BC breaking changes here. So let's try to make everything consistent with `distributed_c10d.py`.